pkg/integrations/prometheus/utils/operatorutils.go

```diff
@@ -0,0 +1,215 @@
+			return
+		}
+		if swapped {
+			finalResult[groupID][timestamp] = valueRHS / valueLHS
+		} else {
+			finalResult[groupID][timestamp] = valueLHS / valueRHS
+		}
+	case utils.LetMultiply:
+		finalResult[groupID][timestamp] = valueLHS * valueRHS
+	case utils.LetSubtract:
+		val := valueLHS - valueRHS
+		if swapped {
+			val = val * -1
+		}
+		finalResult[groupID][timestamp] = val
+	case utils.LetModulo:
+		if swapped {
+			finalResult[groupID][timestamp] = math.Mod(valueRHS, valueLHS)
+		} else {
+			finalResult[groupID][timestamp] = math.Mod(valueLHS, valueRHS)
+		}
+	case utils.LetPower:
+		if swapped {
+			finalResult[groupID][timestamp] = math.Pow(valueRHS, valueLHS)
+		} else {
+			finalResult[groupID][timestamp] = math.Pow(valueLHS, valueRHS)
+		}
+	case utils.LetGreaterThan:
+		isGtr := valueLHS > valueRHS
+		if swapped {
+			isGtr = valueLHS < valueRHS
+		}
+		if isGtr {
+			finalResult[groupID][timestamp] = valueLHS
+		}
+	case utils.LetGreaterThanOrEqualTo:
+		isGte := valueLHS >= valueRHS
+		if swapped {
+			isGte = valueLHS <= valueRHS
+		}
+		if isGte {
+			finalResult[groupID][timestamp] = valueLHS
+		}
+	case utils.LetLessThan:
+		isLss := valueLHS < valueRHS
+		if swapped {
+			isLss = valueLHS > valueRHS
+		}
+		if isLss {
+			finalResult[groupID][timestamp] = valueLHS
+		}
+	case utils.LetLessThanOrEqualTo:
+		isLte := valueLHS <= valueRHS
+		if swapped {
+			isLte = valueLHS >= valueRHS
+		}
+		if isLte {
+			finalResult[groupID][timestamp] = valueLHS
+		}
+	case utils.LetEquals:
+		if valueLHS == valueRHS {
+			finalResult[groupID][timestamp] = valueLHS
+		}
+	case utils.LetNotEquals:
+		if valueLHS != valueRHS {
+			finalResult[groupID][timestamp] = valueLHS
+		}
+	}
+
+	// Logical ops can only been used between 2 vectors
+	if !queryOp.ConstantOp && IsLogicalOperator(queryOp.Operation) {
+		switch queryOp.Operation {
+		case utils.LetAnd:
+			finalResult[groupID][timestamp] = valueLHS
+		case utils.LetOr:
+			finalResult[groupID][timestamp] = valueLHS
+		case utils.LetUnless:
+			finalResult[groupID][timestamp] = valueLHS
+		}
+	}
+}
+
+func ExtractMatchingLabelSet(groupIDStr string, matchingLabels []string, on bool) string {
+
+	labelSetMap := make(map[string]string)
+
+	re, err := regexp.Compile(`(.*)\{(.*)`)
+	if err != nil {
+		return groupIDStr
+	}
+
+	labelSetStr := ""
+	match := re.FindStringSubmatch(groupIDStr)
+	if len(match) == 3 {
+		labelSetStr = match[2]
+	} else {
+		return groupIDStr
+	}
+
+	re, err = regexp.Compile(`\s*([\w\s]+):\s*([\w\s]+)`)
+	if err != nil {
+		return labelSetStr
+	}
+
+	matches := re.FindAllStringSubmatch(labelSetStr, -1)
+	for _, match := range matches {
+		if len(match) == 3 {
+			labelKey := match[1]
+			labelVal := match[2]
+			labelSetMap[labelKey] = labelVal
+		}
+	}
+
+	matchingLabelValStr := ""
+	if on {
+		for _, label := range matchingLabels {
+			val, exists := labelSetMap[label]
+			if exists {
+				matchingLabelValStr += val
+			}
+		}
+	} else { // exclude matchingLabels
```

Comment:
The meaning of this `on` variable isn't clear to me. Perhaps it should be renamed